language/stdlib/modules/LibraAccount.move

```diff
@@ -679,21 +679,32 @@ module LibraAccount {
         new_account_address: address,
         auth_key_prefix: vector<u8>,
     ) {
-        DesignatedDealer::assert_account_is_blessed(blessed);
+        Association::assert_account_is_blessed(blessed);
         Transaction::assert(!Libra::is_synthetic_currency<CoinType>(), 202);
         let new_dd_account = create_signer(new_account_address);
+        Event::publish_generator(&new_dd_account);
         let dealer =
             DesignatedDealer::create_designated_dealer();
-        Event::publish_generator(&new_dd_account);
         make_account<CoinType, DesignatedDealer::Dealer>(new_dd_account, auth_key_prefix, dealer, false)
     }
 
+    /// Create Preburn resource under account 'address' - should be called
+    /// in tandem with 'create_designated_dealer' during DD creation
+    public fun create_preburn_for_address<CoinType>(
+        blessed: &signer,
+        address: address,
+    ) {
+        let account_signer = create_signer(address);
+        Libra::publish_preburn_to_account<CoinType>(blessed, &account_signer);
```

Comment:
I think we can simplify things a bit by just adding this line inside `create_dd_account`. This will make sure that  the only way to get `Preburn` resource is inside a DD account at the time it is created.